standard/expressions.md

```diff
@@ -5901,6 +5948,26 @@ When a property or indexer declared in a *struct_type* is the target of an assig
 >
 > *end example*
 
+### §deconstructing-assignment-new-clause Deconstructing assignment
+
+If the left operand of a `=` operator is classified as a tuple, the right hand side is ***deconstructed*** into individual expressions, which are assigned to each of the elements of the left hand side tuple.
+
+If any of the tuple elements of the left operand have an identifier, a compile-time error occurs. If any of the tuple elements of the left operand is a declaration expression, and any other element is not a declaration expression or discard, a compile-time error occurs.
+
+The evaluation of a deconstructing assignment `(R1, ..., Rn) = E` proceeds as follows:
+
+- First the left hand side tuple is evaluated, meaning that each of the element expressions `R1, ..., Rn` are evaluated in order.
+- Then the right hand side `E` is deconstructed into a list of expressions `E1, E2, ...` as follows:
+  - If `E` is a tuple expression with the same arity as the left-hand side, then `E1, E2, ...` shall be the element expressions of `E`.
+  - Otherwise, if `E` has a tuple type (§tuple-types-new-clause) with the same arity as the left-hand side, then `E1, E2, ...` shall be the member access expressions `E.Item1, E.Item2, ...`, except that `E` shall be evaluated only once.
+  - Otherwise, if `E.Deconstruct(out var v1, out var v2, ...)` with the number of arguments corresponding to the arity of the left hand side is a valid instance method invocation, then that invocation shall be performed and `E1, E2, ...` shall be the simple name expressions `v1, v2, ...`.
+  - Otherwise `E` cannot be deconstructed, and a compile time error occurs.
+- Finally, the operation is evaluated as `(R1 = E1, R2 = E2, ...)`, except that `R1, R2, ...` have already been evaluated and are not evaluated again. This means that individual element-wise assignments are recursively performed, and a resulting tuple value is created.
+
+Note that even though the assignment operation "evaluated as" a tuple expression containing element-wise assignment expressions, this does prevent the deconstructing assignment from occurring as a statement expression, nor the elements on the left-hand side from being declaration expressions.
```

Comment:
I'm not sure I understand this. Should it be "this doesn't prevent..." or should the "nor" be an "or"?

Reply:
The former. Fixed in upcoming commit.